app/components/containers/containersController.js

```diff
@@ -60,48 +60,59 @@ angular.module('containers', [])
 
   var update = function (data) {
     $('#loadContainersSpinner').show();
-    var userDetails = Authentication.getUserDetails();
-    $scope.user = userDetails;
-    $scope.state.selectedItemCount = 0;
-    Container.query(data, function (d) {
-      var containers = d;
-      if ($scope.containersToHideLabels) {
-        containers = ContainerHelper.hideContainers(d, $scope.containersToHideLabels);
-      }
-      $scope.containers = containers.map(function (container) {
-        var model = new ContainerViewModel(container);
-        model.Status = $filter('containerstatus')(model.Status);
+    // Get current endpoint Publish URL
+    var currentEndpointId = EndpointProvider.endpointID();
+    EndpointService.endpoints()
+    .then(function success(endpointsData) {
+      var currentEndpoint = endpointsData.filter(function(e) {
+        return e.Id === currentEndpointId;
+      })[0];
+      var userDetails = Authentication.getUserDetails();
+      $scope.user = userDetails;
+      $scope.state.selectedItemCount = 0;
+      Container.query(data, function (d) {
+        var containers = d;
+        if ($scope.containersToHideLabels) {
+          containers = ContainerHelper.hideContainers(d, $scope.containersToHideLabels);
+        }
+        $scope.containers = containers.map(function (container) {
+          var model = new ContainerViewModel(container, currentEndpoint);
```

Comment:
See below, the container model should not be tied to endpoint data. I think that the URLPublish should be stored in the controller scope (state object) and used in directly the view.